### PR TITLE
Do Not Require `domain=(None, None)` For String Dimensions

### PR DESCRIPTION
**Pull request by nguyenv**

### CODE REVIEW

File: tiledb/libtiledb.pyx [hunk 1]
```diff
@@ -2551,6 +2548,9 @@ cdef class Dim(object):
             # coordinates are written.
             dim_datatype = TILEDB_STRING_ASCII
         else:
+            if domain is None or len(domain) != 2:
```
Comment (ihnorton) [inline tiledb/libtiledb.pyx:2551]: We need to handle `str` / `"U*"` dtypes as well - this only works for `bytes`.

Comment (ihnorton) [reply tiledb/libtiledb.pyx:2551]: (also IMO `domain=(None,None)` is still _valid_ just not required, so let's keep the `len` check too)

**Review by ihnorton — CHANGES_REQUESTED** [review]

File: tiledb/libtiledb.pyx (hunk 1 above)
Comment (nguyenv) [reply tiledb/libtiledb.pyx:2551]: OK I see.

**Review by ihnorton — APPROVED** [review]
LGTM, thanks

**Pull request merged**